pkg/pool-weighted/contracts/managed/ManagedPoolSwapFees.sol

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-interfaces/contracts/vault/IBasePool.sol";
+
+import "./ManagedPoolStorageLib.sol";
+
+abstract contract ManagedPoolSwapFees is IBasePool {
```

Comment:
It's sort of annoying how this makes the contract abstract and greatly increases complexity in the mock.

What if we didn't declare any of the external functions, and left that up for the ManagedPool? We'd then have no need for this module to be able to access state directly (because it is only ever called internally by someone who provides it)